vault/capabilities.go

```diff
@@ -0,0 +1,46 @@
+package vault
+
+import "fmt"
+
+// Capabilities is used to fetch the capabilities of the given token on the given path
+func (c *Core) Capabilities(token, path string) ([]string, error) {
+	if path == "" {
+		return nil, fmt.Errorf("missing path")
+	}
+
+	if token == "" {
+		return nil, fmt.Errorf("missing token")
+	}
+
+	te, err := c.tokenStore.Lookup(token)
+	if err != nil {
+		return nil, err
+	}
+	if te == nil {
+		return nil, fmt.Errorf("invalid token")
```

Comment:
Used errwrap here. Not using any prefix in the errwrap format intentionally. I felt that it is not necessary. Since there were only three error messages, I did not introduce constants for them too.

Please suggest if there is a better way to do this.
errwrap is definitely better than returning two errors for a single call.